link.go

```diff
@@ -9,4 +12,20 @@ type InterfaceLinkDefinition struct {
 	Interfaces   []string          `json:"interfaces,omitempty"`
 	SourceConfig map[string]string `json:"source_config,omitempty"`
 	TargetConfig map[string]string `json:"target_config,omitempty"`
+	// Serialized & encrypted secrets. Should decrypt + deserialize into map[string]SecretValue
+	SourceSecrets []byte `json:"source_secrets,omitempty"`
+	TargetSecrets []byte `json:"target_secrets,omitempty"`
+}
```

Comment:
based on comments in the links PR in wasmCloud, this may be better as a pointer to a byte vec which we can check for nil